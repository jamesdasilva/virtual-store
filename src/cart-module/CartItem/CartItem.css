.cart-item {
  display: flex;
  position: relative;
  height: 100px;
  padding: 10px 0;
  border-top: 1px solid #ddd;
}
.cart-item__thumb {
  box-sizing: content-box;
  position: relative;
  width: 100px;
  flex-shrink: 0;
  flex-grow: 0;
  border: 1px solid #ccc;
  border-radius: 10px;
  overflow: hidden;
}
.cart-item__thumb img {
  width: 100%;
  margin: 0;
}
.cart-item:hover .cart-item__close {
  visibility: visible;
}
.cart-item__close {
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  top: 0;
  left: 0;
  visibility: hidden;
  width: 20px;
  height: 20px;
  background-color: #ff0000;
  cursor: pointer;
}
.cart-item__close div{
  transform: rotate(45deg);
  color: #fff;
  font-weight: bold;
  font-size: 22px
}
.cart-item__info {
  padding: 2px 0 0 8px;
}
.cart-item__name {
  margin: 0;
  color: #78BE20;
  font-size: 12px;
  font-weight: normal;
}
.cart-item__content {
  display: flex;
  justify-content: space-between;
  padding: 6px 0 0 0;
}
.cart-item__code {
  margin: 0 0 6px 0;
  font-size: 11px;
  color: #ccc;
}
.cart-item__info-left {
  display: flex;
  flex-direction: column;
}
.cart-item__info-right {
  display: flex;
  flex-direction: column;
}
.cart-item__from-price {
  margin: 0 0 6px 0;
  font-size: 11px;
  text-align: right;
  color: #ccc;
}
.cart-item__to-price {
  margin: 0;
  font-size: 16px;
  line-height: 16px;
  text-align: right;
  font-weight: bold;
  color: #222;
}