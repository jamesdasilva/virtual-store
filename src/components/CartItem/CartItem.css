.cart-item {
  display: flex;
  height: 100px;
  padding: 10px 0;
  border-top: 1px solid #ddd;
}
.cart-item__thumb {
  box-sizing: content-box;
  width: 100px;
  flex-shrink: 0;
  flex-grow: 0;
  border: 1px solid #ccc;
  border-radius: 10px;
  overflow: hidden;
}
.cart-item__thumb img {
  width: 100%;
  margin: 0;
}
.cart-item__info {
  padding: 2px 0 0 8px;
}
.cart-item__name {
  margin: 0;
  color: #78BE20;
  font-size: 12px;
  font-weight: normal;
}
.cart-item__content {
  display: flex;
  justify-content: space-between;
  padding: 6px 0 0 0;
}
.cart-item__code {
  margin: 0 0 6px 0;
  font-size: 11px;
  color: #ccc;
}
.cart-item__info-left {
  display: flex;
  flex-direction: column;
}
.cart-item__info-right {
  display: flex;
  flex-direction: column;
}
.cart-item__from-price {
  margin: 0 0 6px 0;
  font-size: 11px;
  text-align: right;
  color: #ccc;
}
.cart-item__to-price {
  margin: 0;
  font-size: 16px;
  line-height: 16px;
  text-align: right;
  font-weight: bold;
  color: #222;
}