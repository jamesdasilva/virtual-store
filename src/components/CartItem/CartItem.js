import React from 'react'
import './CartItem.css'
import QuantityField from '../QuantityField/QuantityField'

class CartItem extends React.Component {
  render () {
    return (
      <div className='cart-item'>
        <div className='cart-item__thumb'>
          <img src='https://cdn.leroymerlin.com.br/products/furadeira_parafusadeira_de_impacto_3_8_a_bateria_de_12_volts_1566684694_4ea1_300x300.jpg' />
        </div>
        <div className='cart-item__info'>
          <h3 className='cart-item__name'>Fog√£o Brastemp 4 bocas embutir cor Inox com dupla chama e grill 110V</h3>
          <div className='cart-item__content'>
            <div className='cart-item__info-left'>
              <div className='cart-item__code'>Cod. 9999999</div>
              <QuantityField />
            </div>
            <div className='cart-item__info-right'>
              <div className='cart-item__from-price'>1 un. 999,00</div>
              <div className='cart-item__to-price'>1 un. 777,00</div>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default CartItem
