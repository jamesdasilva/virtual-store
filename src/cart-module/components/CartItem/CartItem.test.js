import '@testing-library/jest-dom'
import { render, cleanup, waitForElement, fireEvent } from '@testing-library/react'
import React from 'react'
import CartItem from './CartItem'

let renderQuantityFieldSpy
let deleteItemSpy
let itemMock

afterEach(cleanup)

beforeEach(() => {
  renderQuantityFieldSpy = jest.fn()
  deleteItemSpy = jest.fn()
  itemMock = { 
    amount: 3,
    product: {
      id: 123,
      name: 'Test Name',
      picture: 'http://uri.com/mock',
      price: {
        from: {integers: "2.849", decimals: "90"},
        to: {integers: "1.849", decimals: "90"}
      }
    }
  }
})

describe('<CartItem /> spec', () => {
  test('Snapshot test', async () => {
    const { getByTestId } = render(
      <CartItem
        renderQuantityField={renderQuantityFieldSpy}
        item={itemMock}
        deleteItem={deleteItemSpy} />
      )
    const cartItem = await waitForElement(() => getByTestId('cart-item'))
    expect(cartItem).toMatchSnapshot()
  })

  test('Should display name, code, image and prices', async () => {    
    const { getByText } = render(
      <CartItem
        renderQuantityField={renderQuantityFieldSpy}
        item={itemMock}
        deleteItem={deleteItemSpy} />
      )

    const productName = await waitForElement(() => getByText(itemMock.product.name))
    expect(productName).toBeDefined()

    const productId = await waitForElement(() => getByText(`Cod. ${itemMock.product.id}`))
    expect(productId).toBeDefined()

    const productPriceFrom = await waitForElement(
      () => getByText(`${itemMock.product.price.from.integers},${itemMock.product.price.from.decimals}`))
    expect(productPriceFrom).toBeDefined()
    
    const productPriceTo = await waitForElement(
      () => getByText(`${itemMock.product.price.to.integers},${itemMock.product.price.to.decimals}`))
    expect(productPriceTo).toBeDefined()
  })

  test('Should call the render function', async () => {
    render(
      <CartItem
        renderQuantityField={renderQuantityFieldSpy}
        item={itemMock}
        deleteItem={deleteItemSpy} />
      )
    expect(renderQuantityFieldSpy).toBeCalled()
  })

  test('Should call deleteItem function when user clicks icon close', async () => {
    const { getByTestId } = render(
      <CartItem
        renderQuantityField={renderQuantityFieldSpy}
        item={itemMock}
        deleteItem={deleteItemSpy} />
      )
    const cartItemClose = await waitForElement(() => getByTestId('cart-item-close'))
    fireEvent.click(cartItemClose)
    expect(deleteItemSpy).toBeCalled()
  })

})
