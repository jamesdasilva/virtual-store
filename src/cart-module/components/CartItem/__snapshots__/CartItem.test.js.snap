// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem /> spec Snapshot test 1`] = `
<div
  class="cart-item"
  data-testid="cart-item"
>
  <div
    class="cart-item__thumb"
  >
    <div
      class="cart-item__close"
      data-testid="cart-item-close"
    >
      <div>
        +
      </div>
    </div>
    <img
      src="http://uri.com/mock"
    />
  </div>
  <div
    class="cart-item__info"
  >
    <h3
      class="cart-item__name"
    >
      Test Name
    </h3>
    <div
      class="cart-item__content"
    >
      <div
        class="cart-item__info-left"
      >
        <div
          class="cart-item__code"
        >
          Cod. 
          123
        </div>
      </div>
      <div
        class="cart-item__info-right"
      >
        <div
          class="cart-item__from-price"
        >
          2.849,90
        </div>
        <div
          class="cart-item__to-price"
        >
          1.849,90
        </div>
      </div>
    </div>
  </div>
</div>
`;
